src/native/crashtracker.rs

```diff
@@ -0,0 +1,283 @@
+use std::collections::HashMap;
+use std::ops::DerefMut;
+use std::sync::atomic::{AtomicU8, Ordering};
+use std::sync::Once;
+
+use datadog_crashtracker::{
+    CrashtrackerConfiguration, CrashtrackerReceiverConfig, Metadata, StacktraceCollection,
+};
+use ddcommon::Endpoint;
+use pyo3::prelude::*;
+
+// We redefine the Enum here to expose it to Python as datadog_crashtracker::StacktraceCollection
+// is defined in an external crate.
+#[pyclass(
+    eq,
+    eq_int,
+    name = "StacktraceCollection",
+    module = "datadog.internal._native"
+)]
+#[derive(Clone, PartialEq)]
+pub enum StacktraceCollectionPy {
+    Disabled,
+    WithoutSymbols,
+    EnabledWithInprocessSymbols,
+    EnabledWithSymbolsInReceiver,
+}
+
+impl From<StacktraceCollectionPy> for StacktraceCollection {
+    fn from(value: StacktraceCollectionPy) -> Self {
+        match value {
+            StacktraceCollectionPy::Disabled => StacktraceCollection::Disabled,
+            StacktraceCollectionPy::WithoutSymbols => StacktraceCollection::WithoutSymbols,
+            StacktraceCollectionPy::EnabledWithInprocessSymbols => {
+                StacktraceCollection::EnabledWithInprocessSymbols
+            }
+            StacktraceCollectionPy::EnabledWithSymbolsInReceiver => {
+                StacktraceCollection::EnabledWithSymbolsInReceiver
+            }
+        }
+    }
+}
+
+#[pyclass(
```

Comment:
Try moving this into libdatadog under `--feature pyo3`, this could help removing most of the boilerplate code we have here